fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCQueryBuilder.java

```diff
@@ -1384,4 +1386,242 @@ private SqlQueryData processMissingParm(Class<?> resourceType, QueryParameter qu
+            nextChainedLogicalResourceVar = CLR + (parmIndex + 2);
+            nextChainedParmVar        = CP + (parmIndex + 2);
+            StringBuilder whereClauseSegment = new StringBuilder();
+
+            if (parmIndex == 0) {
+                // BUild outer select: SELECT CLR0.LOGICAL_ID
+                //                       FROM <resource-type>_LOGICAL_RESOURCES AS CLR0
+                //                       JOIN <resource-type>_RESOURCES AS CR0
+                //                         ON CR0.RESOURCE_ID = CLR0.CURRENT_RESOURCE_ID AND CR0.IS_DELETED = 'N'
+                //                       WHERE
+                selectSegments.append("SELECT " + prevChainedLogicalResourceVar + DOT + "LOGICAL_ID")
+                                .append(FROM + resourceType.getSimpleName() + "_LOGICAL_RESOURCES AS " + prevChainedLogicalResourceVar)
+                                .append(JOIN + resourceType.getSimpleName() + "_RESOURCES AS " + prevChainedResourceVar)
+                                .append(ON + prevChainedResourceVar + DOT + "RESOURCE_ID" + EQ + prevChainedLogicalResourceVar + DOT + "CURRENT_RESOURCE_ID")
+                                .append(AND + prevChainedResourceVar + DOT + "IS_DELETED" + EQ + "'N'")
+                                .append(WHERE);
+            }
+
+            if (parmIndex < lastParmIndex) {
+                if (currentParm.isReverseChained()) {
+                    // Build inner select joins: EXISTS (SELECT 1
+                    //                           FROM <modifierTypeResourceName>_TOKEN_VALUES_V AS CPx
+                    //                           JOIN <modifierTypeResourceName>_LOGICAL_RESOURCES AS CLRx
+                    //                             ON CLRx.LOGICAL_RESOURCE_ID = CPx.LOGICAL_RESOURCE_ID
+                    //                           JOIN <modifierTypeResourceName>_RESOURCES AS CRx
+                    //                             ON CRx.RESOURCE_ID = CLRx.CURRENT_RESOURCE_ID AND CRx.IS_DELETED = 'N'
+                    selectSegments.append("EXISTS " + LEFT_PAREN + "SELECT 1")
+                                    .append(FROM + currentParm.getModifierResourceTypeName() + "_TOKEN_VALUES_V AS " + chainedParmVar)
+                                    .append(JOIN + currentParm.getModifierResourceTypeName() + "_LOGICAL_RESOURCES AS " + chainedLogicalResourceVar)
+                                    .append(ON + chainedLogicalResourceVar + DOT + "LOGICAL_RESOURCE_ID" + EQ + chainedParmVar + DOT + "LOGICAL_RESOURCE_ID")
+                                    .append(JOIN + currentParm.getModifierResourceTypeName() + "_RESOURCES AS " + chainedResourceVar)
+                                    .append(ON + chainedResourceVar + DOT + "RESOURCE_ID" + EQ + chainedLogicalResourceVar + DOT + "CURRENT_RESOURCE_ID")
+                                    .append(AND + chainedResourceVar + DOT + "IS_DELETED" + EQ + "'N'");
```

Comment:
same comment about appends and the EXISTS I think is already in the JDBCConstants

same with RESOURCE_ID and IS_DELETED

Reply:
done